examples/secured-client-with-oauth2/secured_client_with_oauth2.bal

```diff
@@ -2,30 +2,101 @@ import ballerina/http;
-            clientId: "506144513496-dqm5vdqfrfhdjjom10rmvafb8e3h7rtm.apps.googleusercontent.com",
-            clientSecret: "3hw2XN4MfiIRrv6mghX6m5gM",
-            refreshToken: "1/UwH3YyYccKTrH9bqj35Y7hMYTK9f3HEC3uzlrleFwPE",
-            refreshUrl: "https://www.googleapis.com/oauth2/v4/token"
+// OAuth2 authentication is enabled by setting the `scheme: http:OAUTH2`.
+// OAuth2 client credentials grant type is configured by setting
+// `grantType: http:CLIENT_CREDENTIALS_GRANT`.
+http:Client clientEP1 = new("https://api.bitbucket.org/2.0", config = {
+        auth: {
+            scheme: http:OAUTH2,
+            config: {
+                grantType: http:CLIENT_CREDENTIALS_GRANT,
+                config: {
+                    tokenUrl: "https://bitbucket.org/site/oauth2/access_token",
+                    clientId: "mMNWS9PLmM93V5WHjC",
+                    clientSecret: "jLY6xPY3ER4bNTspaGu6fb7kahhs7kUa"
+                }
+            }
         }
-    }
-});
+    });
+
+// Define the OAuth2 client endpoint to call the backend services.
+// OAuth2 authentication is enabled by setting the `scheme: http:OAUTH2`.
+// OAuth2 password grant type is configured by setting
+// `grantType: http:PASSWORD_GRANT`.
+// If access-token get expired or invalid the it will be automatically
+// refreshed with the provided `refreshUrl`.
+http:Client clientEP2 = new("https://api.bitbucket.org/2.0", config = {
+        auth: {
+            scheme: http:OAUTH2,
+            config: {
+                grantType: http:PASSWORD_GRANT,
+                config: {
+                    tokenUrl: "https://bitbucket.org/site/oauth2/access_token",
+                    username: "b7a.demo@gmail.com",
+                    password: "39^R3ODx",
+                    clientId: "mMNWS9PLmM93V5WHjC",
+                    clientSecret: "jLY6xPY3ER4bNTspaGu6fb7kahhs7kUa",
+                    refreshConfig: {
+                        refreshUrl: "https://bitbucket.org/site/oauth2/access_token"
+                    }
+                }
+            }
+        }
+    });
+
+// Define the OAuth2 client endpoint to call the backend services.
+// OAuth2 authentication is enabled by setting the `scheme: http:OAUTH2`.
+// OAuth2 direct token mode is configured by setting
+// `grantType: http:DIRECT_TOKEN`.
+// If `accessToken` is invalid or not provided, it will be automatically
+// refreshed with the provided `clientId`, `clientSecret`, `refreshToken`,
+// `refreshUrl`.
+http:Client clientEP3 = new("https://www.googleapis.com/tasks/v1", config = {
+        auth: {
+            scheme: http:OAUTH2,
+            config: {
+                grantType: http:DIRECT_TOKEN,
+                config: {
+                    accessToken: "ya29.GlvQBkqJS0yn0zsZm4IIUUzLk3DH1rRiCMKnHiz6deycKmTFiDsuoFlFfrmXF8dCb0gyzLyXpnv3VcrIlauj3nMs61CbydaAqMl6RwVIU2r2qg1StVVvxRWT9_Or",
```

Comment:
Examples should have max char length 80. Please check all the places

Reply:
Yes. But in this case, if I split the tokens into different lines with 80 char length the internally generated payload body of the request also reflects that. So, that leads to a valid request body, but invalid tokens at the OAuth2 server endpoint. Hence the test gets failed.

**Example**: Request for refresh the access token.

This is how the payload is generated if I split the tokens.
```
grant_type=refresh_token&refresh_token=1/UwH3YyYccKTrH9bqj35Y7hMYTK9f3HEC3uzlrl
                        eFwPE  
```

This is how the payload should be according to the spec.
```
grant_type=refresh_token&refresh_token=1/UwH3YyYccKTrH9bqj35Y7hMYTK9f3HEC3uzlrleFwPE  
```

Reply:
A not so nice approach here is to define a string with 80 char limit (string concatenated), and the use it in the configuration. 

```ballerina
string accessToken = "ya29.GlvQBkqJS0yn0zsZm4IIUUzLk3DH1rRiCMKnHiz6deycKm" + 
                      "TFiDsuoFlFfrmXF8dCb0gyzLyXpnv3VcrIlauj3nMs61CbydaAqMl6" + 
                      "RwVIU2r2qg1StVVvxRWT9_Or";

// ...
config: {
    accessToken: accessToken
    // ...
}
```